### PR TITLE
ksuperkey: git-2015-07-21 -> 0.4

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Motivation for this change
update to a stable (and newer) release instead of a random git rev + cleanup

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/X11/ksuperkey/default.nix [hunk 1]
```diff
@@ -1,27 +1,28 @@
-{ stdenv, fetchgit, libX11, libXtst, pkgconfig, xorgproto, libXi }:
+{ stdenv, fetchFromGitHub, libX11, libXtst, pkgconfig, xorgproto, libXi }:
 
 stdenv.mkDerivation {
```
Comment (marsam) [inline pkgs/tools/X11/ksuperkey/default.nix:3]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/tools/X11/ksuperkey/default.nix [hunk 2]
```diff
@@ -1,27 +1,28 @@
...
-  ];
-
-  src = fetchgit {
-    url = "https://github.com/hanschen/ksuperkey";
-    rev = "e75a31a0e3e80b14341e92799a7ce3232ac37639";
-    sha256 = "0y4wkak9dvcm14g54ll1ln9aks2az63hx8fv7b8d3nscxjbkxl6g";
+  src = fetchFromGitHub {
+    owner = "hanschen";
+    repo = "ksuperkey";
+    rev = "v0.4";
```
Comment (marsam) [inline pkgs/tools/X11/ksuperkey/default.nix:10]:
```suggestion
    rev = "v${version}";
```

File: pkgs/tools/X11/ksuperkey/default.nix [hunk 3]
```diff
@@ -1,27 +1,28 @@
...
+    sha256 = "1dvgf356fihfav8pjzww1q6vgd96c5h18dh8vpv022g9iipiwq8a";
   };
 
   preConfigure = ''
-    makeFlags="$makeFlags PREFIX=$out"
+    makeFlags="PREFIX=$out"
   '';
```
Comment (marsam) [inline pkgs/tools/X11/ksuperkey/default.nix:16]: I think you can replace this with:
```
  makeFlags = [ "PREFIX=${placeholder "out"}" ];
```

**Comment by IvarWithoutBones** [comment]
All done :+1:

**Pull request merged**